### PR TITLE
Auto-update deps when --upgrade is provided.

### PR DESCRIPTION
**Pull request by Cynocracy**
This is preferable to requiring devs to manually curate all
dependencies, but has some unfortunate fallout in that we
will need to pin all dependencies with breaking changes
and not just those that are updated by test-infra changes.

See https://github.com/knative/pkg/pull/1463

**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+.cov9 { color: rgb(32, 224, 152) }
+.cov10 { color: rgb(20, 236, 155) }
+
+		</style>
+	</head>
+	<body>
+		<div id="topbar">
+			<div id="nav">
+				<select id="files">
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:56]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+		</style>
+	</head>
+	<body>
+		<div id="topbar">
+			<div id="nav">
+				<select id="files">
+				
+				<option value="file0">knative.dev/test-infra/tools/coverage/testTarget/presubmit/common.go (0.0%)</option>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:58]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+	</head>
+	<body>
+		<div id="topbar">
+			<div id="nav">
+				<select id="files">
+				
+				<option value="file0">knative.dev/test-infra/tools/coverage/testTarget/presubmit/common.go (0.0%)</option>
+				
+				<option value="file1">knative.dev/test-infra/tools/coverage/testTarget/presubmit/cov-excl.go (28.6%)</option>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:60]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+		<div id="topbar">
+			<div id="nav">
+				<select id="files">
+				
+				<option value="file0">knative.dev/test-infra/tools/coverage/testTarget/presubmit/common.go (0.0%)</option>
+				
+				<option value="file1">knative.dev/test-infra/tools/coverage/testTarget/presubmit/cov-excl.go (28.6%)</option>
+				
+				<option value="file2">knative.dev/test-infra/tools/coverage/testTarget/presubmit/ling-gen.go (46.7%)</option>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:62]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+				<select id="files">
+				
+				<option value="file0">knative.dev/test-infra/tools/coverage/testTarget/presubmit/common.go (0.0%)</option>
+				
+				<option value="file1">knative.dev/test-infra/tools/coverage/testTarget/presubmit/cov-excl.go (28.6%)</option>
+				
+				<option value="file2">knative.dev/test-infra/tools/coverage/testTarget/presubmit/ling-gen.go (46.7%)</option>
+				
+				<option value="file3">knative.dev/test-infra/tools/coverage/testTarget/presubmit/newlyAddedFile.go (74.1%)</option>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:64]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+				<option value="file0">knative.dev/test-infra/tools/coverage/testTarget/presubmit/common.go (0.0%)</option>
+				
+				<option value="file1">knative.dev/test-infra/tools/coverage/testTarget/presubmit/cov-excl.go (28.6%)</option>
+				
+				<option value="file2">knative.dev/test-infra/tools/coverage/testTarget/presubmit/ling-gen.go (46.7%)</option>
+				
+				<option value="file3">knative.dev/test-infra/tools/coverage/testTarget/presubmit/newlyAddedFile.go (74.1%)</option>
+				
+				<option value="file4">knative.dev/test-infra/tools/coverage/testTarget/presubmit/onlySrcChange.go (66.7%)</option>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:66]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+				<option value="file1">knative.dev/test-infra/tools/coverage/testTarget/presubmit/cov-excl.go (28.6%)</option>
+				
+				<option value="file2">knative.dev/test-infra/tools/coverage/testTarget/presubmit/ling-gen.go (46.7%)</option>
+				
+				<option value="file3">knative.dev/test-infra/tools/coverage/testTarget/presubmit/newlyAddedFile.go (74.1%)</option>
+				
+				<option value="file4">knative.dev/test-infra/tools/coverage/testTarget/presubmit/onlySrcChange.go (66.7%)</option>
+				
+				<option value="file5">knative.dev/test-infra/tools/coverage/testTarget/presubmit/onlyTestChange.go (66.7%)</option>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:68]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+				
+				<option value="file4">knative.dev/test-infra/tools/coverage/testTarget/presubmit/onlySrcChange.go (66.7%)</option>
+				
+				<option value="file5">knative.dev/test-infra/tools/coverage/testTarget/presubmit/onlyTestChange.go (66.7%)</option>
+				
+				</select>
+			</div>
+			<div id="legend">
+				<span>not tracked</span>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:73]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+				<span class="cov2">*</span>
+				<span class="cov3">*</span>
+				<span class="cov4">*</span>
+				<span class="cov5">*</span>
+				<span class="cov6">*</span>
+				<span class="cov7">*</span>
+				<span class="cov8">*</span>
+				<span class="cov9">*</span>
+				<span class="cov10">high coverage</span>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:85]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+				<span class="cov6">*</span>
+				<span class="cov7">*</span>
+				<span class="cov8">*</span>
+				<span class="cov9">*</span>
+				<span class="cov10">high coverage</span>
+			
+			</div>
+		</div>
+		<div id="content">
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:89]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 11]
```diff
@@ -0,0 +1,332 @@
...
+*/
+
+// Package presubmit serves as the target for presubmit coverage test.
+package presubmit
+
+func doNothing() {<span class="cov10" title="35">
+
+}</span>
+</pre>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:113]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 12]
```diff
@@ -0,0 +1,332 @@
...
+}</span>
+
+func fakeFuncCovExcl_2() <span class="cov8" title="1">{
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+}</span>
+</pre>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:152]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 13]
```diff
@@ -0,0 +1,332 @@
...
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+}</span>
+</pre>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:191]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 14]
```diff
@@ -0,0 +1,332 @@
...
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+        doNothing()
+
+}</span>
+</pre>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:244]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 15]
```diff
@@ -0,0 +1,332 @@
...
+func fakeFunc1_1() <span class="cov8" title="1">{
+        doNothing()
+        doNothing()
+}</span>
+
+func fakeFunc1_2() <span class="cov0" title="0">{
+        doNothing()
+}</span>
+</pre>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:273]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html [hunk 16]
```diff
@@ -0,0 +1,332 @@
...
+func funcOTC_1() <span class="cov8" title="1">{
+        doNothing()
+        doNothing()
+}</span>
+
+func funcOTC_2() <span class="cov0" title="0">{
+        doNothing()
+}</span>
+</pre>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_317343246/line-cov.html:302]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+.cov9 { color: rgb(32, 224, 152) }
+.cov10 { color: rgb(20, 236, 155) }
+
+		</style>
+	</head>
+	<body>
+		<div id="topbar">
+			<div id="nav">
+				<select id="files">
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html:56]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+	<body>
+		<div id="topbar">
+			<div id="nav">
+				<select id="files">
+				
+				</select>
+			</div>
+			<div id="legend">
+				<span>not tracked</span>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html:61]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+				<span class="cov2">*</span>
+				<span class="cov3">*</span>
+				<span class="cov4">*</span>
+				<span class="cov5">*</span>
+				<span class="cov6">*</span>
+				<span class="cov7">*</span>
+				<span class="cov8">*</span>
+				<span class="cov9">*</span>
+				<span class="cov10">high coverage</span>
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html:73]: Remove trailing whitespace:
```suggestion

```

File: tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+				<span class="cov6">*</span>
+				<span class="cov7">*</span>
+				<span class="cov8">*</span>
+				<span class="cov9">*</span>
+				<span class="cov10">high coverage</span>
+			
+			</div>
+		</div>
+		<div id="content">
+
```
Comment (mattmoor) [inline tools/coverage/test_output/tmp_artifacts/TestRunPresubmit_487527292/line-cov.html:77]: Remove trailing whitespace:
```suggestion

```

**Comment by chaodaiG** [comment]
I consider this a good move
/approve

/assign @chizhg

**Pull request merged**